src/cache/prognostic_edmf_precomputed_quantities.jl

```diff
@@ -41,14 +43,93 @@ NVTX.@annotate function set_prognostic_edmf_precomputed_quantities_environment!(
         Y.c.ρ,
         turbconv_model,
     )
+
     set_sgs_ᶠu₃!(u₃⁰, ᶠu₃⁰, Y, turbconv_model)
     set_velocity_quantities!(ᶜu⁰, ᶠu³⁰, ᶜK⁰, ᶠu₃⁰, Y.c.uₕ, ᶠuₕ³)
     # @. ᶜK⁰ += ᶜtke⁰
+
     @. ᶜts⁰ = TD.PhaseEquil_phq(thermo_params, ᶜp, ᶜmse⁰ - ᶜΦ, ᶜq_tot⁰)
     @. ᶜρ⁰ = TD.air_density(thermo_params, ᶜts⁰)
     return nothing
 end
 
+NVTX.@annotate function set_prognostic_edmf_precomputed_quantities_environment!(
+    Y,
+    p,
+    ᶠuₕ³,
+    t,
+    ::NonEquilMoistModel,
+)
+    @assert !(p.atmos.moisture_model isa DryModel)
+
+    thermo_params = CAP.thermodynamics_params(p.params)
+    (; turbconv_model) = p.atmos
+    (; ᶜΦ,) = p.core
+    (; ᶜp, ᶜh_tot, ᶜK) = p.precomputed
+
+    (;
+        ᶜtke⁰,
+        ᶜρa⁰,
+        ᶠu₃⁰,
+        ᶜu⁰,
+        ᶠu³⁰,
+        ᶜK⁰,
+        ᶜts⁰,
+        ᶜρ⁰,
+        ᶜmse⁰,
+        ᶜq_tot⁰,
+        ᶜq_liq⁰,
+        ᶜq_ice⁰,
+    ) = p.precomputed
+
+    @. ᶜρa⁰ = ρa⁰(Y.c)
+    @. ᶜtke⁰ = divide_by_ρa(Y.c.sgs⁰.ρatke, ᶜρa⁰, 0, Y.c.ρ, turbconv_model)
+    @. ᶜmse⁰ = divide_by_ρa(
+        Y.c.ρ * (ᶜh_tot - ᶜK) - ρamse⁺(Y.c.sgsʲs),
+        ᶜρa⁰,
+        Y.c.ρ * (ᶜh_tot - ᶜK),
+        Y.c.ρ,
+        turbconv_model,
+    )
+    @. ᶜq_tot⁰ = divide_by_ρa(
+        Y.c.ρq_tot - ρaq_tot⁺(Y.c.sgsʲs),
+        ᶜρa⁰,
+        Y.c.ρq_tot,
+        Y.c.ρ,
+        turbconv_model,
+    )
+    @. ᶜq_liq⁰ = divide_by_ρa(
+        Y.c.ρq_liq - ρaq_liq⁺(Y.c.sgsʲs),
+        ᶜρa⁰,
+        Y.c.ρq_liq,
+        Y.c.ρ,
+        turbconv_model,
+    )
+    @. ᶜq_ice⁰ = divide_by_ρa(
+        Y.c.ρq_ice - ρaq_ice⁺(Y.c.sgsʲs),
+        ᶜρa⁰,
+        Y.c.ρq_ice,
+        Y.c.ρ,
+        turbconv_model,
+    )
+
+    set_sgs_ᶠu₃!(u₃⁰, ᶠu₃⁰, Y, turbconv_model)
+    set_velocity_quantities!(ᶜu⁰, ᶠu³⁰, ᶜK⁰, ᶠu₃⁰, Y.c.uₕ, ᶠuₕ³)
+    # @. ᶜK⁰ += ᶜtke⁰
+
+    # create phase partition
+    #@. q_pt⁰ = TD.PhasePartition(ᶜq_tot⁰, ᶜq_liq⁰, ᶜq_ice⁰)
+    @. ᶜts⁰ = TD.PhaseNonEquil_phq(
+        thermo_params,
+        ᶜp,
+        ᶜmse⁰ - ᶜΦ,
+        TD.PhasePartition.(ᶜq_tot⁰, ᶜq_liq⁰, ᶜq_ice⁰),
```

Comment:
```suggestion
        nonequil_phase_part(ᶜts, Y.c.sgsʲs, ᶜρa⁰, Y.c.ρ, turbconv_model),
```
and somewhere in the module we can define:

```julia
nonequil_phase_part(ts, sgsʲs, ᶜρa⁰, ρ, turbconv_model) =
    nonequil_phase_part(
        ρ * TD.total_specific_humidity(ts),
        ρ * TD.liquid_specific_humidity(ts),
        ρ * TD.ice_specific_humidity(ts),
        sgsʲs, ᶜρa⁰, ρ, turbconv_model
    )
function nonequil_phase_part(ρq_tot, ρq_liq, ρq_ice, sgsʲs, ᶜρa⁰, ρ, turbconv_model)
    q_tot = divide_by_ρa(
        ρq_tot - ρaq_tot⁺(sgsʲs),
        ᶜρa⁰,
        ρq_tot,
        ρ,
        turbconv_model,
    )
    q_liq = divide_by_ρa(
        ρq_liq - ρaq_liq⁺(sgsʲs),
        ᶜρa⁰,
        ρq_liq,
        ρ,
        turbconv_model,
    )
    q_ice = divide_by_ρa(
        ρq_ice - ρaq_ice⁺(sgsʲs),
        ᶜρa⁰,
        ρq_ice,
        ρ,
        turbconv_model,
    )
return TD.PhasePartition(q_tot, q_liq, q_ice)
end
```